internal/mode/static/state/graph/gatewayclass.go

```diff
@@ -75,11 +76,27 @@ func buildGatewayClass(gc *v1beta1.GatewayClass) *GatewayClass {
 	}
 }
 
-func validateGatewayClass(gc *v1beta1.GatewayClass) error {
+func validateGatewayClass(gc *v1beta1.GatewayClass, npCfg *ngfAPI.NginxProxy) error {
 	if gc.Spec.ParametersRef != nil {
-		path := field.NewPath("spec").Child("parametersRef")
-		return field.Forbidden(path, "parametersRef is not supported")
+		if _, ok := supportedParamKinds[string(gc.Spec.ParametersRef.Kind)]; !ok {
+			path := field.NewPath("spec").Child("parametersRef")
+			return field.Forbidden(path, "parametersRef resource not allowed")
+		}
+
+		if gc.Spec.ParametersRef.Namespace == nil {
+			path := field.NewPath("spec").Child("parametersRef")
+			return field.Required(path, "parametersRef namespace must be specified for NginxProxy")
+		}
+
+		if npCfg == nil {
```

Comment:
Just to double check we're ok with it.
This means if NginxProxy doesn't exist (for example, it was removed for some reason) no data plane configuration will be generated.

cc @kate-osborn

Reply:
No data plane config will be generated for whatever fields are in the NginxProxy config, yes.